### ISSUE TITLE
Respond 408 status when the server didn't receive request fully until it's timed out

### ISSUE THREAD

**Issue by minwoox** [line/armeria#5579]
Currently, we always respond with 503 if a `RequestTimeoutException` is raised.
However, we should send 408 status if the server didn't receive the request fully.

```
The 408 (Request Timeout) status code indicates that the server did not receive a complete request message within the time that it was prepared to wait.
```
https://httpwg.org/specs/rfc9110.html#status.408

Related: https://github.com/line/armeria/issues/5572

**Comment by thachlp** [comment]
Can I work on this?

**Comment by thachlp** [comment]
> Hi, @thachlp! We are currently addressing some timeout-related issues, so I believe this task should be completed after that. I will inform you once the timeout issues have been resolved.

Thanks

**Comment by minwoox** [comment]
@thachlp, Sorry I had to do this by myself because it's more complicated than I initially thought. 😓

**Comment by thachlp** [comment]
> @thachlp, Sorry I had to do this by myself because it's more complicated than I initially thought. 😓

It is fine, thank you for information 🙇

**Comment by minwoox** [comment]
Closed by #5680
